### PR TITLE
Allow publishing of decrypted packets to MQTT

### PR DESCRIPTION
**Pull request by joshpirihi**
The MQTT functionality is a great way to get data out of the mesh and into external systems, particularly with a custom mqtt broker.  However the encryption makes this difficult.

This PR adds the option to publish decrypted packets to MQTT, depending on the user preferences mqtt_server and mqtt_encryption_enabled.

Note that there is a small change in default behaviour with this PR:  If a user has set a custom mqtt broker, then by default the device will publish decrypted packets.  This will have the effect of making it easier for people who want to plot their environmental plugin sensors in grafana (for example), and should not affect mesh bridging over mqtt.

### CODE REVIEW

File: src/mesh/Router.cpp [hunk 1]
```diff
@@ -210,14 +210,43 @@ ErrorCode Router::send(MeshPacket *p)
...
+         *    not set                        1              1
+         *        set                        0              0
+         *        set                        1              1
+         * 
+         * => so we only encrypt mqtt if they have a custom mqtt server AND mqtt_encryption_enabled is FALSE
+         */ 
+
+        bool shouldActuallyEncrypt = true;
+        if (*radioConfig.preferences.mqtt_server && !radioConfig.preferences.mqtt_encryption_enabled) {
+            shouldActuallyEncrypt = false;
```
Comment (mc-hamster) [inline src/mesh/Router.cpp:230]: Our configuration settings should always default to false and we should avoid changing existing behavior where possible. 

Request here is to change this to mqtt_encryption_disabled where the default is 0.

**Review by mc-hamster — CHANGES_REQUESTED** [review]
Thanks for the PR. Requesting one change.

File: src/mesh/Router.cpp (hunk 1 above)
Comment (garthvh) [reply src/mesh/Router.cpp:230]: I would agree with this in general, but I have not found a single user who has managed to decrypt the encrypted messages.

Comment (joshpirihi) [reply src/mesh/Router.cpp:230]: Totally happy to change it, but I'll just let this conversation play out for a little while first

Comment (mc-hamster) [reply src/mesh/Router.cpp:230]: This is used by our existing MQTT public server and hub. It's also part of the spec:

https://meshtastic.org/docs/software/other/mqtt

If we change this behavior, the spec should also be updated.

Comment (joshpirihi) [reply src/mesh/Router.cpp:230]: It won't affect the existing public server - the preference is ignored if the user has not set a mqtt_server (and thus connects to mqtt.meshtastic.org).  In that case it's encrypted anyway.

That docs page isn't very representative of the current implementation (looks more like a brainstorm?).  I'll have a go at adding a section that explains how it actually works right now

Comment (mc-hamster) [reply src/mesh/Router.cpp:230]: @joshpirihi I totally missed that in my review. Both mqtt server and the new conf has to be set.

Thanks for hearing this out. PR is accepted.

**Review by mc-hamster — APPROVED** [review]

File: src/mesh/Router.cpp (hunk 1 above)
Comment (joshpirihi) [reply src/mesh/Router.cpp:230]: awesome! thanks for your feedback, always appreciated!

**Comment by mc-hamster** [comment]
thanks!

**Pull request merged**